### PR TITLE
Migrate/to nextcloud vue 8

### PR DESCRIPTION
**Pull request by GVodyanov**
Fix https://github.com/nextcloud/calendar/issues/5619

Progress:

- [x] Change title to name
- [x] Change MultiSelect to Select props
- [x] Work out errors with Select when `multiple = true`

### CODE REVIEW

**Comment by hamza221** [comment]
pushed a commit here by mistake hence the push -f 
NcSelect fix here with this branch as base #5649

**Comment by ChristophWurst** [comment]
What is left to do here?

**Comment by GVodyanov** [comment]
Other issue found by Christoph, deleted calendars aren't striked through, will work on

**Review by ChristophWurst — COMMENTED** [review]
:+1: tested & works

My only remark would be the NcSelects that have dynamic width. This makes the UI look jittery.

Bildschirmfoto vom 2024-01-29 09-56-18
Bildschirmfoto vom 2024-01-29 09-56-05
Bildschirmfoto vom 2024-01-29 09-55-47

NcSelect also seems to increase it's height when it is focused/open. Upstream bug or our styling?

**Comment by ChristophWurst** [comment]
- [ ] BUG when I'm searching for attendees

```
[Vue warn]: Error in render: "TypeError: option is undefined"

found in

---> <VueSelect>
       <NcSelect>
         <InviteesListSearch> at src/components/Editor/Invitees/InviteesListSearch.vue
           <InviteesList> at src/components/Editor/Invitees/InviteesList.vue
             <NcAppSidebarTab>
               <NcAppSidebarTabs>
                 <NcAppSidebar>
                   <EditSidebar> at src/views/EditSidebar.vue
                     <NcContent>
                       <Calendar> at src/views/Calendar.vue
                         <App> at src/App.vue
                           <Root> vue.runtime.esm.js:4609
```

**Review by st3iny — COMMENTED** [review]
I tested a bit and couldn't spot any additional breakage.

I pushed some commits to fix multiple minor issues in the sidebar editor:
- Vertical alignment
- Select widths overflowing on small screens
- Hide the title submit button again

**Comment by GVodyanov** [comment]
> * [x]  BUG when I'm searching for attendees
> 
> 
> ```
> [Vue warn]: Error in render: "TypeError: option is undefined"
> 
> found in
> 
> ---> <VueSelect>
>        <NcSelect>
>          <InviteesListSearch> at src/components/Editor/Invitees/InviteesListSearch.vue
>            <InviteesList> at src/components/Editor/Invitees/InviteesList.vue
>              <NcAppSidebarTab>
>                <NcAppSidebarTabs>
>                  <NcAppSidebar>
>                    <EditSidebar> at src/views/EditSidebar.vue
>                      <NcContent>
>                        <Calendar> at src/views/Calendar.vue
>                          <App> at src/App.vue
>                            <Root> vue.runtime.esm.js:4609
> ```

Fixed

**Comment by ChristophWurst** [comment]
- [ ] BUG room type is unusable

image

```
[Vue warn]: Error in render: "TypeError: option is undefined"

found in

---> <VueSelect>
       <NcSelect>
         <ResourceRoomType> at src/components/Editor/Resources/ResourceRoomType.vue
           <ResourceListSearch> at src/components/Editor/Resources/ResourceListSearch.vue
             <ResourceList> at src/components/Editor/Resources/ResourceList.vue
               <NcAppSidebarTab>
                 <NcAppSidebarTabs>
                   <NcAppSidebar>
                     <EditSidebar> at src/views/EditSidebar.vue
                       <NcContent>
                         <Calendar> at src/views/Calendar.vue
                           <App> at src/App.vue
                             <Root> vue.runtime.esm.js:4609
TypeError: option is undefined
    fn ResourceRoomType.vue:25
    VueJS 2
    fn NcSelect.mjs:430
    VueJS 2
    node_modules calendar-main.js:2269
    renderList VueJS
    node_modules calendar-main.js:2269
    VueJS 14
    onSearchFocus vue-select.js:1
    VueJS 33
vue.runtime.esm.js:3049
```

**Comment by st3iny** [comment]
> I still get a build warning
> 
> ```
> WARNING in ./src/components/AppNavigation/Settings/SettingsAttachmentsFolder.vue?vue&type=script&lang=js (./node_modules/babel-loader/lib/index.js!./node_modules/vue-loader/lib/index.js??vue-loader-options!./src/components/AppNavigation/Settings/SettingsAttachmentsFolder.vue?vue&type=script&lang=js) 8:4-14
> export 'NcInputField' (imported as 'InputField') was not found in '@nextcloud/vue' (possible exports: Focus, Linkify, MOBILE_BREAKPOINT, NcActionButton, NcActionButtonGroup, NcActionCaption, NcActionCheckbox, NcActionInput, NcActionLink, NcActionRadio, NcActionRouter, NcActionSeparator, NcActionText, NcActionTextEditable, NcActions, NcAppContent, NcAppContentDetails, NcAppContentList, NcAppNavigation, NcAppNavigationCaption, NcAppNavigationIconBullet, NcAppNavigationItem, NcAppNavigationNew, NcAppNavigationNewItem, NcAppNavigationSettings, NcAppNavigationSpacer, NcAppSettingsDialog, NcAppSettingsSection, NcAppSidebar, NcAppSidebarTab, NcAutoCompleteResult, NcAvatar, NcBreadcrumb, NcBreadcrumbs, NcButton, NcCheckboxRadioSwitch, NcColorPicker, NcContent, NcCounterBubble, NcDashboardWidget, NcDashboardWidgetItem, NcDateTime, NcDateTimePicker, NcDateTimePickerNative, NcDialog, NcDialogButton, NcEmojiPicker, NcEmptyContent, NcGuestContent, NcHeaderMenu, NcHighlight, NcIconSvgWrapper, NcListItem, NcListItemIcon, NcLoadingIcon, NcMentionBubble, NcModal, NcNoteCard, NcPasswordField, NcPopover, NcProgressBar, NcRelatedResourcesPanel, NcRichContenteditable, NcRichText, NcSavingIndicatorIcon, NcSelect, NcSelectTags, NcSettingsInputText, NcSettingsSection, NcSettingsSelectGroup, NcTextArea, NcTextField, NcTimezonePicker, NcUserBubble, NextcloudVuePlugin, Tooltip, clickOutsideOptions, emojiAddRecent, emojiSearch, isA11yActivation, isFullscreen, isFullscreenState, isMobile, isMobileState, richEditor, useIsFullscreen, useIsMobile, userStatus, usernameToColor)
> ```

Ahh, this one. The component has to be imported directly from `nextcloud/Vue/dist/Components/NcInputField.vue` or something similar.

It is not included in the main bundle as we are not supposed to use it directly.

**Comment by GVodyanov** [comment]
> nextcloud/Vue/dist/Components/NcInputField.vue

Could you please specify exactly how to import it and where? I'm a bit confused

File: src/components/AppNavigation/Settings/SettingsAttachmentsFolder.vue [hunk 1]
```diff
@@ -37,15 +37,16 @@
 </template>
 
 <script>
+import { NcInputField as InputField } from '@nextcloud/vue'
+
 import debounce from 'debounce'
 import { mapState } from 'vuex'
 import { getFilePickerBuilder, showError, showSuccess } from '@nextcloud/dialogs'
-import NcInputField from '@nextcloud/vue/dist/Components/NcInputField.js'
```
Comment (ChristophWurst) [inline src/components/AppNavigation/Settings/SettingsAttachmentsFolder.vue:43]: this is the way it should be imported

**Comment by ChristophWurst** [comment]
- [ ] BUG: recurrence end selector

Bildschirmfoto vom 2024-02-06 12-21-11

**Comment by ChristophWurst** [comment]
- [ ] BUG (minor): *Clear selection* shows for the calendar picker that always has a value

Bildschirmfoto vom 2024-02-06 12-21-50

**Comment by GVodyanov** [comment]
> * [x]  BUG: recurrence end selector
> 
> 
> Bildschirmfoto vom 2024-02-06 12-21-11 Bildschirmfoto vom 2024-02-06 12-21-09

Leaving this comment here for future reference as this is a bit weird and it took me a while to understand: for some reason there is an internal css variable of `vue-select` called `--vs-dropdown-min-width` and it is set to `160px`. Attempting to set an `NcSelect` to anything thinner than that will result in this bug, and it's hard to trace as this is internal styling of `vue-select`.

**Comment by GVodyanov** [comment]
> * [ ]  BUG (minor): _Clear selection_ shows for the calendar picker that always has a value
> 
> 
> Bildschirmfoto vom 2024-02-06 12-21-50

I'm a little blocked here, an input event is sent out with a value of an empty array but the value of the NcSelect doesn't change, could someone else take a look at this please?

**Comment by GVodyanov** [comment]
`PropertySelectMultiple` isn't working as intended, options aren't being removed from the dropdown when selected and an error is thrown, currently working on that.

**Comment by GVodyanov** [comment]
✨ _Other NcSelect bugs_ ✨
- There is no way to remove categories in `PropertySelectMultiple`
    - Only way to solve this is to make a custom close icon because the default one doesn't appear seeing as we embed a custom component
- You can add the same category more than one time

**Comment by ChristophWurst** [comment]
- [ ] BUG: shared calendars as rendered as two lines once picked in the calendar picker

| Selecting | Selected |
|--------|--------|
| Bildschirmfoto vom 2024-02-19 13-12-21 | Bildschirmfoto vom 2024-02-19 13-12-25 | 

Ignore the broken avatar. That is an issue with my dev env.

**Review by ChristophWurst — COMMENTED** [review]
:+1: otherwise. tested and works :)

**Review by st3iny — APPROVED** [review]
Tested and works. Didn't have a look at the code.

**Comment by GVodyanov** [comment]
> * [ ]  BUG: alignment of appointments section and + icon
> 
> 
> v4.5 	Here
> Bildschirmfoto vom 2024-02-19 13-10-25 	Bildschirmfoto vom 2024-02-19 13-10-18
> 
> **Richard:** I can't reproduce this one.

Could you please suggest a way to reproduce? I can't get it to show either. @ChristophWurst

**Comment by st3iny** [comment]
Well then time to resolve conflicts and squash commits @GVodyanov. Let me know if you need help with the conflicts.

**Pull request merged**